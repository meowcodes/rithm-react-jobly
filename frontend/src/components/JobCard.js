import React, { Component } from 'react';
import './styles/JobCard.css';

/**Recieves ONE job data
 * Renders a card with job data*/
class JobCard extends Component {

    shouldComponentUpdate(nextProps, nextState){
        return nextProps.applied !== this.props.applied;
    }

    render() {
        // check if already applied (in currUser.jobs )
        const { title, salary, equity, applied } = this.props;
        const button = (applied
            ? <button disabled>APPLIED</button> 
            : <button onClick={ this.props.triggerApply }>APPLY</button>
        );

        return (
            <div className="JobCard">
                <div className="JobCard-text">
                    <p className="JobCard-title">
                        {title}
                    </p>

                    <p className="JobCard-salary">
                        Salary: {salary}
                    </p>

                    <p className="JobCard-equity">
                        Equity: {equity}
                    </p>
                </div>
                { button }
            </div>
        );
    }
}

export default JobCard;
