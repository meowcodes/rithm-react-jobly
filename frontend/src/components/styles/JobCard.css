.JobCard {
  min-width: 300px;
  max-width: 480px;
  margin   : 1.5rem auto;
  position : relative;
}

.JobCard {
  border    : solid 1px gainsboro;
  padding   : 1rem;
  text-align: left;
}

.JobCard:hover {
  box-shadow: 0px 0px 3px gainsboro;
}

.JobCard button {
  position: absolute;
  bottom  : 0.75rem;
  right   : 1.25rem;
}

.JobCard button:hover {
  background: transparent;
  border    : solid 1px turquoise;
  box-shadow: 0px 0px 3px turquoise;
}

.JobCard-text {
  display: inline-block;
}