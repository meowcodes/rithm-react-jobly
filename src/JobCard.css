.JobCard, 
.JobCard-link {
  width: 600px;
  margin: 1.5em auto;
  position: relative;
}

.JobCard {
  border: solid 1px gainsboro;
  padding: 1em;
  text-align: left;
}

.JobCard:hover {
  box-shadow: 0px 0px 3px turquoise;
}

.JobCard button {
  position: absolute;
  margin: 2em;
  bottom: 1em;
  right: 0;
}

.JobCard-text {
    display: inline-block;
  }
