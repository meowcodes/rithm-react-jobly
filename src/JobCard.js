import React, { Component } from 'react';
import './JobCard.css';
/**
 * Recieves ONE job data
 * Renders a card with job data
 */

class JobCard extends Component {

  render() {
    const { title, salary, equity } = this.props;
    return (
        <div className="JobCard">
            <div className="JobCard-text">
                <p className="JobCard-title">
                    { title }
                </p>

                <p className="JobCard-salary">
                    Salary: { salary }
                </p>

                <p className="JobCard-equity">
                    Equity: { equity }
                </p>
            </div>
            <button>APPLY</button>

        </div>
    );
  }
}

export default JobCard;
