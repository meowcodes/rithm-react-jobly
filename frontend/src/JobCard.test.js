import React from 'react';
import { mount, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { MemoryRouter } from 'react-router-dom';
import JobCard from './JobCard';


// full mount
it('mounts without crashing', function() {
  mount(
    <MemoryRouter>
      <JobCard 
        title="test" 
        salary={100}
        equity={0.10}
        applied="applied"/>
    </MemoryRouter>
  );
});

// snapshot test
it('matches snapshot', function() {
  let wrapper = shallow(<JobCard 
                          title="test" 
                          salary={100}
                          equity={0.10}
                          applied="applied"/>);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});
