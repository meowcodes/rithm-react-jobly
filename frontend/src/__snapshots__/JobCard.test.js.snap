// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="JobCard"
>
  <div
    className="JobCard-text"
  >
    <p
      className="JobCard-title"
    >
      test
    </p>
    <p
      className="JobCard-salary"
    >
      Salary: 
      100
    </p>
    <p
      className="JobCard-equity"
    >
      Equity: 
      0.1
    </p>
  </div>
  <button
    disabled={true}
  >
    APPLY
  </button>
</div>
`;
